src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -129,95 +96,37 @@ public FlinkPravegaReader(final URI controllerURI, final String scope, final Set
-     *
-     * @param checkpointInitiateTimeout The timeout, in milliseconds
-     */
-    public void setCheckpointInitiateTimeout(long checkpointInitiateTimeout) {
-        Preconditions.checkArgument(checkpointInitiateTimeout > 0, "timeout must be >= 0");
-        this.checkpointInitiateTimeout = checkpointInitiateTimeout;
-    }
-
-    /**
-     * Gets the timeout for initiating a checkpoint in Pravega.
-     *
-     * <p>This timeout if applied to the future returned by
-     * {@link io.pravega.client.stream.ReaderGroup#initiateCheckpoint(String, ScheduledExecutorService)}.
-     *
-     * @return The timeout, in milliseconds
-     */
-    public long getCheckpointInitiateTimeout() {
-        return checkpointInitiateTimeout;
-    }
-
-    /**
-     * Gets the timeout for the call to read events from Pravega. After the timeout
-     * expires (without an event being returned), another call will be made.
-     * 
-     * <p>This timeout is passed to {@link EventStreamReader#readNextEvent(long)}.
-     * 
-     * @param eventReadTimeout The timeout, in milliseconds
-     */
-    public void setEventReadTimeout(long eventReadTimeout) {
-        Preconditions.checkArgument(eventReadTimeout > 0, "timeout must be >= 0");
-        this.eventReadTimeout = eventReadTimeout;
+    protected FlinkPravegaReader(String hookUid, ClientConfig clientConfig,
+                                 ReaderGroupConfig readerGroupConfig, String readerGroupScope, String readerGroupName,
+                                 DeserializationSchema<T> deserializationSchema, Time eventReadTimeout, Time checkpointInitiateTimeout) {
+
+        this.hookUid = Preconditions.checkNotNull(hookUid, "hookUid");
+        this.clientConfig = Preconditions.checkNotNull(clientConfig, "clientConfig");
+        this.readerGroupConfig = Preconditions.checkNotNull(readerGroupConfig, "readerGroupConfig");
+        this.readerGroupScope = Preconditions.checkNotNull(readerGroupScope, "readerGroupScope");
+        this.readerGroupName = Preconditions.checkNotNull(readerGroupName, "readerGroupName");
+        this.deserializationSchema = Preconditions.checkNotNull(deserializationSchema, "deserializationSchema");
+        this.eventReadTimeout = Preconditions.checkNotNull(eventReadTimeout, "eventReadTimeout");
+        this.checkpointInitiateTimeout = Preconditions.checkNotNull(checkpointInitiateTimeout, "checkpointInitiateTimeout");
     }
 
     /**
-     * Gets the timeout for the call to read events from Pravega.
-     * 
-     * <p>This timeout is the value passed to {@link EventStreamReader#readNextEvent(long)}.
-     * 
-     * @return The timeout, in milliseconds
+     * Initializes the reader.
      */
-    public long getEventReadTimeout() {
-        return eventReadTimeout;
+    void initialize() {
+        // TODO: This will require the client to have access to the pravega controller and handle any temporary errors.
+        //       See https://github.com/pravega/pravega/issues/553.
```

Comment:
This issue is actually closed. If this is still a concern, then we need an issue against this repository.

I'm wondering if this is an actual concern, though. On the one hand, if the controller is not accessible when this environment is being constructed, then it is a good idea to fail early. On the other hand, this is just setting up configuration for submission, and creating a reader group is really part of runtime. 

It also depends on when this `initialize` call is actually invoked. If at job submission or job execution time, then we are fine and we don't need another issue.

Reply:
That's fine, but let's at least replace the issue from the pravega queue, which is currently closed, with one in the flink-connector queue. I have created this one:

https://github.com/pravega/flink-connectors/issues/130